### PR TITLE
[FIX JENKINS-37753] REGRESSION: Steps showing up as incomplete when they are in fact complete

### PR DESCRIPTION
**Pull request by scherler**
# Description

See [JENKINS-37753](https://issues.jenkins-ci.org/browse/JENKINS-37753).

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [X] Appropriate unit or acceptance tests or explanation to why this change has no tests
https://github.com/jenkinsci/blueocean-acceptance-test/pull/43 -> not tested due to problems with updatecenter mirror in spain, @recena is looking into it. -> NOW tested since update center is back
- [x] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [x] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@jenkinsci/code-reviewers @reviewbybees 

…inished with the run, this makes sure that they will be refetched after the run ended.

### CODE REVIEW

**Comment by recena** [comment]
The [mirror](http://mirrors.clinkerhq.com/jenkins/) is back again.

File: blueocean-dashboard/src/main/js/components/RunDetailsPipeline.jsx [hunk 1]
```diff
@@ -176,6 +176,31 @@ export class RunDetailsPipeline extends Component {
...
+                    if (keyArray[keyArray.length - 1] === 'nodes') {
+                        keyArray.pop();
+                    }
+                    const searchString = keyArray.join('/');
+                    // fire to remove all logs from cache
+                    removeLogs(searchString);
+                    /** we have to check now if all steps have been declared as finished,
+                     * if not we will remove them from the cache so we get the final information*/
+                    const notFinishedSteps = Object.keys(this.props.steps)
+                        .filter((item) => item.indexOf(searchString) !== -1)
```
Comment (kzantow) [inline blueocean-dashboard/src/main/js/components/RunDetailsPipeline.jsx]: NIT: isn't this condition usually written as `>= 0` ? or should I adopt a different standard?

Comment (scherler) [inline blueocean-dashboard/src/main/js/components/RunDetailsPipeline.jsx]: @kzantow talk to @tfennelly about that. 

I did it in another PR like you said and then @tfennelly ask the same question as you but the other way around. I personally do not care, but having this NIT in all PR is getting old. ;)

Comment (kzantow) [inline blueocean-dashboard/src/main/js/components/RunDetailsPipeline.jsx]: Too funny. :) Not a reason to hold it up at all, obviously.

Comment (kzantow) [inline blueocean-dashboard/src/main/js/components/RunDetailsPipeline.jsx]: Mozilla examples refer to this method, I suppose I should try to adapt my brain: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/indexOf

File: blueocean-dashboard/src/main/js/redux/actions.js [hunk 2]
```diff
@@ -140,12 +142,24 @@ export const actionHandlers = {
         steps[payload.nodesBaseUrl] = payload;
         return state.set('steps', steps);
     },
+    [ACTION_TYPES.REMOVE_STEP](state, { payload }): State {
```
Comment (kzantow) [inline blueocean-dashboard/src/main/js/redux/actions.js]: :ant: `payload` is entirely nondescript as to what the data contains

**Comment by michaelneale** [comment]
lint failures now...

File: blueocean-dashboard/src/main/js/redux/actions.js (hunk 2 above)
Comment (scherler) [inline blueocean-dashboard/src/main/js/redux/actions.js]: Well that is still in my head from my last job. We used it always as "envelope" for the data. This way we had only few attributes in our commands/events, however there is no reason to keep doing it I guess.

I took care of it.

**Comment by sophistifunk** [comment]
Goddammit, is this the same as https://github.com/jenkinsci/blueocean-plugin/pull/521

**Comment by scherler** [comment]
@sophistifunk not really. here we making sure that the steps and logs are getting cleaned when the run has finished

**Comment by sophistifunk** [comment]
I'll see myself out :)

**Comment by scherler** [comment]
@sophistifunk but an excellent way to catch attention to your PR though ;) jeje

File: blueocean-dashboard/src/main/js/components/RunDetailsPipeline.jsx (hunk 1 above)
Comment (michaelneale) [inline blueocean-dashboard/src/main/js/components/RunDetailsPipeline.jsx]: huh

**Comment by michaelneale** [comment]
Looks like Keith as used the github approval, which is as good as a :bee: ;)

**Pull request merged**